### PR TITLE
General edit of new plugins documentation

### PR DESCRIPTION
**Pull request by krisgesling**
Extend on PR #184 - "Adds information on how a module is created to be detected as a Mycroft plugin (as proposed in mycroft-core PR #2594) and some basic instructions on how to create the different kind of plugins."

Also includes general copy edit and added pages to Gitbook navigation (SUMMARY.md)

New PR so that Gitbook doesn't modify anything between the two commits.

### CODE REVIEW

**Comment by krisgesling** [comment]
@forslund just in case you want to check the edits before it gets pushed

File: docs/mycroft-technologies/mycroft-core/plugins/tts.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    "module": "example_tts",
+    "example_tts": {
+      ...
+    }
+  }
+```
+
+Mycroft will register the `"example_tts"` part in the TTS's `self.config`
+
+### `get_tts()`
```
Comment (forslund) [inline docs/mycroft-technologies/mycroft-core/plugins/tts.md:41]: Should probably be updated to 
```
## `get_tts()`
```

File: docs/mycroft-technologies/mycroft-core/plugins/tts.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+- `sentence` (str): a piece of text to turn into audio.
+- `wav_file` (str): path to where Mycroft should store the generated audio data.
+
+This method should generate audio data and return a Tuple:
+- path to written data (generally the input argument)
+- viseme list for synthesized audio
+
+As an example see the [`get_tts` method for Mimic2](https://github.com/MycroftAI/mycroft-core/blob/dev/mycroft/tts/mimic2_tts.py#L225).
+
+### TTS Validator
```
Comment (forslund) [inline docs/mycroft-technologies/mycroft-core/plugins/tts.md:53]: Same as above

File: docs/mycroft-technologies/mycroft-core/plugins/tts.md [hunk 3]
```diff
@@ -40,17 +40,19 @@ Mycroft will register the `"example_tts"` part in the TTS's `self.config`
 
 ### `get_tts()`
 
-The `get_tts()` method will be called by Mycroft to generate audio and (optionally) phonemes. This is the main method that the plugin creator needs to implement.
+The `get_tts()` method will be called by Mycroft to generate audio and (optionally) phonemes. This is the main method that the plugin creator needs to implement. It is called with:
+- `sentence` (str): a piece of text to turn into audio.
+- `wav_file` (str): path to where Mycroft should store the generated audio data.
```
Comment (forslund) [inline docs/mycroft-technologies/mycroft-core/plugins/tts.md:45]: where the plugin should store the generated audio data.

File: docs/mycroft-technologies/mycroft-core/plugins/tts.md [hunk 4]
```diff
@@ -40,17 +40,19 @@ Mycroft will register the `"example_tts"` part in the TTS's `self.config`
...
 
-It's called with a `sentence`, a piece of text to turn into audio, and a `wav_file` an path to where Mycroft would like to store the generated audio data.
+This method should generate audio data and return a Tuple:
+- path to written data (generally the input argument)
+- viseme list for synthesized audio
```
Comment (forslund) [inline docs/mycroft-technologies/mycroft-core/plugins/tts.md:49]: MIght be better to format this as a tuple with the parenthesis and coma to be clear

**Comment by forslund** [comment]
Thanks for looking it over and cleaning it up. Noted a couple of small things I saw.

**Comment by krisgesling** [comment]
Great spotting. Fixes are pushed. Hopefully the tuple return is clearer now.

**Pull request merged**